pl_examples/loop_examples/kfold.py

```diff
@@ -216,13 +216,13 @@ def on_load_checkpoint(self, state_dict: Dict) -> None:
         self.current_fold = state_dict["current_fold"]
 
     def _reset_fitting(self) -> None:
-        self.trainer.reset_train_dataloader()
```

Comment:
not that it's private, is there a way for the user to manually reset/load the dataloader?
One case I was using this one is to prepare an lr_scheduler in configure_optimizers since dataloaders aren't loaded at that point.

Reply:
Using grep.app I only found two cases of users using `reset_train/val/test/predict_dataloader`, so I thought it might be okay to deprecate and make private. But if we still want to users to have this functionality, I can move them to the Trainer

Reply:
yeah! I think it's pretty useful.